src/components/WorkOrders/WorkOrdersView.js

```diff
@@ -38,6 +39,31 @@ const WorkOrdersView = ({ query }) => {
     updateUrlQueryParams({})
   }
 
+  const onFilterRemove = (category, indexNumber) => {
+    const updatedAppliedFilters = convertValuesOfObjectToArray(queryParams, [
+      'pageNumber',
+      'IncludeHistorical',
+    ])
+    switch (category.toLowerCase()) {
+      case 'contractor':
+        updatedAppliedFilters.ContractorReference.splice(indexNumber, 1)
+        break
+      case 'priority':
+        updatedAppliedFilters.Priorities.splice(indexNumber, 1)
+        break
+      case 'status':
+        updatedAppliedFilters.StatusCode.splice(indexNumber, 1)
+        break
+      case 'trade':
+        updatedAppliedFilters.TradeCodes.splice(indexNumber, 1)
+        break
+    }
+
+    updateUrlQueryParams(updatedAppliedFilters)
+    delete updatedAppliedFilters['pageNumber']
```

Comment:
Why do we need to do this?

Reply:
I was trying to stay consistent with the logic of passing the same values (https://github.com/LBHackney-IT/repairs-hub-frontend/blob/a3cf733ca6ebd8cc28bf3c4bd751f310c1997389/src/components/WorkOrders/WorkOrdersView.js#L25) 
to `setAppliedFilters` we always pass object of arrays with filter values 
thats the only reason